hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/catalyst/catalog/HoodieCatalogTable.scala

```diff
@@ -313,7 +315,9 @@ class HoodieCatalogTable(val spark: SparkSession, var table: CatalogTable) exten
           KeyGeneratorType.valueOf(originTableConfig(HoodieTableConfig.KEY_GENERATOR_TYPE.key)).getClassName)
     } else {
       val primaryKeys = table.properties.getOrElse(SQL_KEY_TABLE_PRIMARY_KEY.sqlKeyName, table.storage.properties.get(SQL_KEY_TABLE_PRIMARY_KEY.sqlKeyName)).toString
-      val partitions = table.partitionColumnNames.mkString(",")
+      val partitionFieldsOpt = Option.apply(ConfigUtils.getStringWithAltKeys(originTableConfig.asJava.asInstanceOf[java.util.Map[String, Object]], HoodieTableConfig.PARTITION_FIELDS))
```

Comment:
`originTableConfig.asJava.asInstanceOf[java.util.Map[String, Object]` may incur non-trivial overhead.  Could you add a new util method that can be used on Scala Map?

Reply:
I made an attempt to avoid it. I created a new function in ConfigUtils which uses a key mapping function and used that instead in Scala.

Reply:
I prefer the older approach. LMK, I can revert the last commit if you feel the same.